src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1183,873 @@ def test_rotate_img_metadata(self):
+
+
+class TestTintToColormap(unittest.TestCase):
+    # input tuple RGB, colormap or str
+    # output tint as colormap
+    # test for different input types
+    # test for TypeError/ValueError
+
+    def test_tint_to_colormap_colormap(self):
+        """
+        Test if an input of a known colormap as tint will return the same colormap
+        """
+        input_tint = cm.get_cmap('viridis')
+
+        output_colormap = img.tintToColormap(input_tint)
+
+        self.assertEqual(output_colormap, input_tint)
+
+    def test_tint_to_colormap_tuple(self):
+        """
+        Test if an input of a tuple tint will return a LinearSegmentedColormap
+        """
+        input_tint = (0, 73, 255)
+
+        output_colormap = img.tintToColormap(input_tint)
+
+        self.assertIsInstance(output_colormap, cm.colors.LinearSegmentedColormap)
+
+    def test_tint_to_colormap_known_str(self):
+        """
+        Test if an input of a known str tint will return a
+        LinearSegmentedColormap or a ListedColormap
+        """
+        input_tint = TINT_RGB_AS_IS
+
+        output_colormap = img.tintToColormap(input_tint)
+
+        self.assertIsInstance(output_colormap, cm.colors.LinearSegmentedColormap)
+
+        input_tint = TINT_FIT_TO_RGB
+
+        output_colormap = img.tintToColormap(input_tint)
+
+        self.assertIsInstance(output_colormap, cm.colors.ListedColormap)
+
+    def test_tint_to_colormap_unexpected_input(self):
+        """
+        Test if a wrong tint type will raise a TypeError and a
+        wrong tint value will raise a ValueError
+        """
+        input_tint = "norgb"
+
+        with self.assertRaises(TypeError):
+            img.tintToColormap(input_tint)
+
+        input_tint = (0, 300, 255)
+
+        with self.assertRaises(ValueError):
+            img.tintToColormap(input_tint)
+
+
+class TestGetYXFromZYX(unittest.TestCase):
+    # input 3D data array as picture index int for z-axis start
+    # output 2D data array
+    # test for single dimension
+    # test forcing TypeError/ValueError
+    # test for 2D pos
+
+    def setUp(self):
+        # create a mock grayscale 3D image array
+        size = (200, 399, 399)
+        self.md = {
+              model.MD_DIMS: 'YXZ',
+              model.MD_PIXEL_SIZE: (1e-6, 1e-6),
+              model.MD_BPP: 8,
+              model.MD_POS: (5.0, 7.0)
+             }
+
+        # create a mock 3D image array with white color values
+        self.arr1 = numpy.ones(size, dtype='uint8') * 255
+
+        x_count, y_count = 0, 0
+        self.color_num = 160
+
+        # change values of the ndarray in a pyramid shape
+        # so that every z-slice contains a different amount
+        # of gray color values
+        for z in range(self.arr1.shape[0]):
+            for y in range(y_count, self.arr1.shape[1] - y_count):
+                for x in range(x_count, self.arr1.shape[2] - x_count):
+                    self.arr1[z, y, x] = self.color_num
```

Comment:
Typically, the last loop, you'd write as a single line, like this:
```python
                    self.arr1[z, y, x_count:self.arr1.shape[2] - x_count] = self.color_num
```
It does the same, just a lot faster. That's not a big deal in a test case, but good to know.